segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DataFrameReader.java

```diff
@@ -45,6 +48,10 @@
     @Setter (AccessLevel.PROTECTED)
     private long lastReadSequenceNumber;
     private boolean closed;
+    private final String traceObjectId;
```

Comment:
DataFrameReader does not appear to be thread safe. If that OK and intended it should be annotated as @NotThreadSafe

Reply:
Yes, I think it is intended because is supposed to be executed by one thread upon bootstrap. But yes, we can add that annotation to the class to make it explicit.